### PR TITLE
[W5.3b][W11-A2] Li JiaYao

### PR DESCRIPTION
**Pull request by JYL123**
Extract commonalities from Phone, Email and Address classes into a parent class called Contact. 

Contact class has fields: value, isPrivate
Contact class has methods: isValidContact, toString, equals,  hashCode, isPrivate

### CODE REVIEW

File: src/seedu/addressbook/data/person/Contact.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
+package seedu.addressbook.data.person;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+/**
+ * A superclass for Email, Address, Phone in the addressbook.
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Contact.java]: no need to mention about the sub-classes here. just adding information on what this class is about is sufficient.

File: src/seedu/addressbook/data/person/Contact.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+ * Implementations should guarantee: value is present and isPrivate is provided.
+ * is valid as declared in {@link #isValidContact(String, String)}
+ */
+public class Contact {
+
+    private String value;
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Contact.java]: :+1:

File: src/seedu/addressbook/data/person/Contact.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    @Override
+    public String toString() {
+        return value;
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+                || (other instanceof Contact // instanceof handles nulls
+                && this.value.equals(((Contact) other).value)); // state check
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Contact.java]: This can potentially equate a phone number with address, think why?

**Comment by okkhoy** [comment]
@JYL123 
Some comments added. Please ack & close the PR after reading comments.

File: src/seedu/addressbook/data/person/Contact.java (hunk 3 above)
Comment (JYL123) [reply src/seedu/addressbook/data/person/Contact.java]: Oh, as they are both objects of type Contact, so not just for address and phone number, but also for email, address, phone too. Then this is not a good abstraction. Thanks!

**Comment by JYL123** [comment]
Changes are made. Thanks for reviewing. PR is closed.

**Pull request closed without merging**